date_module.js

```diff
@@ -0,0 +1,88 @@
+//setdate
+// set a variable for the date using built in js date function
+// setDate is a method that allows the value of the date variable to be changed by user input
+// user enters a new date which is saved to variable
+
+var myObj = ( function iife() {
+  var myDate = new Date();
+
+  function setDate() {
+    var userInputDate;
+    if (typeof userInputDate !== 'undefined') {
+      myDate = userInputDate;
+    }
+  }
+
+  function getDate( format ) {
+  	myDate.getDate({ format: "milliseconds" });
```

Comment:
How this function should work is:
1. instantiate the internally stored date if it hasn't been yet
2. check if the user provided an argument to the method call - i.e. check if "format" has a value set and if it does, if it has the property "format" set on it.
3. If format is an object with a "format" property set on it, check the value that was passed for that property - if that value was "millisecond", return the millisecond representation of the internally stored date. If "formatted" was the value set for "format", return a string with the following, example format: "April 25, 2016". 
4. If no argument was passed in, default to returning the millisecond value of the internally stored date.